CoordgenMinimizer.cpp

```diff
@@ -108,7 +109,7 @@ bool CoordgenMinimizer::applyForces(float maxd)
 /*store extra interaction to be used when minimizing molecule.
  cis amides constraints are an example as they need 3d coordinates
  to be detected*/
-void CoordgenMinimizer::addExtraInteraction(
+void EXPORT_COORDGEN CoordgenMinimizer::addExtraInteraction(
```

Comment:
only declarations need to be done - that is, only things in header files, since that's how we expect clients to use this. A version of the rules is here: https://gcc.gnu.org/wiki/Visibility (gcc) or https://developer.apple.com/library/content/documentation/DeveloperTools/Conceptual/CppRuntimeEnv/Articles/SymbolVisibility.html (apple) or https://docs.microsoft.com/en-us/cpp/build/exporting-from-a-dll-using-declspec-dllexport (windows)

Same comments for other files cpp files.